### PR TITLE
Generic JSON read/write type

### PR DESCRIPTION
**Pull request by Krule**
Hi, following suggestion on Gitter, https://gitter.im/rom-rb/chat?at=597fb669210ac269204184f4
I would like to propose a generic json read/write type: `Coercible::Json` together with:

- `Coercible::Json()`
- `Coercible::JsonHash()`
- `Coercible::HashJson()`

### CODE REVIEW

**Comment by Krule** [comment]
Also, if there is something wrong with the approach I have used, pointers would be appreciated. 😄

File: core/lib/rom/types.rb [hunk 1]
```diff
@@ -22,5 +25,30 @@ def ForeignKey(relation, type = Types::Int)
         type.meta(foreign_key: true, target: relation)
       end
     end
+
+    module CoercibleMethods
+      def JsonRead(symbol_keys = false, type = Types::Hash)
+        Types.Constructor(type) do |value|
+          next Hash[value] if value.respond_to?(:to_hash)
```
Comment (flash-gordon) [inline core/lib/rom/types.rb]: I don't think we need this check in ROM 4.0, `value` is expected to string and we don't process it twice. I hope 😬 However, in 3.0 it's required for rom-repo

Comment (Krule) [inline core/lib/rom/types.rb]: @flash-gordon So, maybe a different PR for 4.0 and getting this one into 3.X?

Comment (Krule) [inline core/lib/rom/types.rb]: Here it is: https://github.com/rom-rb/rom/pull/446

**Comment by flash-gordon** [comment]
Thanks!

**Comment by solnic** [comment]
Please rename `Json` to `JSON`

**Pull request merged**